source/MaterialXRenderGlsl/GLTextureHandler.cpp

```diff
@@ -208,62 +214,50 @@ bool GLTextureHandler::bindImage(const FilePath& filePath, const ImageSamplingPr
     return false;
 }
 
-int GLTextureHandler::mapAddressModeToGL(ImageSamplingProperties::AddressMode addressModeEnum)
+bool GLTextureHandler::unbindImage(const FilePath& filePath)
 {
-    const vector<int> addressModes 
-    {
-        // Constant color. Use clamp to border
-        // with border color to achieve this
-        GL_CLAMP_TO_BORDER,
-
-        // Clamp
-        GL_CLAMP_TO_EDGE,
-
-        // Repeat
-        GL_REPEAT,
-
-        // Mirror
-        GL_MIRRORED_REPEAT
-    };
-
-    int addressMode = GL_REPEAT;
-    if (addressModeEnum != ImageSamplingProperties::AddressMode::UNSPECIFIED)
+    const ImageDesc* cachedDesc = getCachedImage(filePath);
+    if (cachedDesc)
     {
-        addressMode = addressModes[static_cast<int>(addressModeEnum)];
+        return unbindImage(*cachedDesc);
     }
-    return addressMode;
+    return false;
 }
 
-int GLTextureHandler::mapFilterTypeToGL(ImageSamplingProperties::FilterType filterTypeEnum)
+bool GLTextureHandler::unbindImage(const ImageDesc& imageDesc)
```

Comment:
New unbind. Pulled out from delete as it should be available when the shader is not not using texture anymore, not just when deleting, otherwise all texures would have to be reloaded all the time.